### PR TITLE
refactor: [M3-7588] - Remove `classnames` and `@types/classnames`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Removes `classnames` (and `@types/classnames`) 🧹
  - `tss-react` contains the same helper (cx)
  - Streamlines how we handle conditional classnames 
  - I could be wrong, and the difference would be minimal, but using cx over classnames may improve performance because I think it uses the emotion cache?
  - I'm really just trying to find ways to continue to lighten up our dependencies

## How to test 🧪

- Check for UI regressions in touched components 👁️

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/PrimaryNav/NavItem.tsx [hunk 1]
```diff
@@ -51,17 +53,14 @@ export const NavItem = React.memo((props: Props) => {
     <React.Fragment>
       {href ? (
         <Link
-          className={classNames({
-            [linkClasses(href)]: true,
-            listItemCollapsed: isCollapsed,
```
Comment (bnussman-akamai) [inline packages/manager/src/components/PrimaryNav/NavItem.tsx:56]: `listItemCollapsed` isn't a thing so I removed it

**Review by dwiley-akamai — COMMENTED** [review]
I need to finish looking through the app but the Selects, gravatars, and notifications/events are looking good so far

**Review by abailly-akamai — APPROVED** [review]
nice cleanup! this was an old lib that has been there from almost day one i think 👍 

visually tested all components affected against prod and found no regressions ✅

**Review by jaalah-akamai — APPROVED** [review]
Yea that's true - essentially a 1:1 swap here 🧹

**Review by dwiley-akamai — APPROVED** [review]
No regressions observed during my visual audit of the app ✅
Code review ✅

**Pull request merged**